test/profiler/test_cpp_thread.py

```diff
@@ -188,7 +193,142 @@ def test_without_enable_profiler_in_child_thread(self) -> None:
         IS_WINDOWS,
         "Failing on windows cuda, see https://github.com/pytorch/pytorch/pull/130037 for slightly more context",
     )
-    def test_profile_memory(self) -> None:
+    def test_profile_memory_cuda(self) -> None:
+        self.start_profiler(True)
+        cpp.start_threads(self.ThreadCount, IterationCount, True)
+        self.check_trace(
+            {
+                "aten::add": [self.ThreadCount, "CPU"],
+            },
+            mem=True,
+        )
+
+
+# Here duplicate the CppThreadTestXPU to enable the xpu cases because
+# the instantiate_device_type_tests will call setUp to launch the threads
+# unexpectedly
+class CppThreadTestXPU(TestCase):
```

Comment:
Cannot share the code with `CPPThreadTestCUDA` because the `instantiate_device_type_tests` will call class method `setUpClass`. In function `setUpClass`, the instantiated class(e.g `CppThreadTestCPU`, `CppThreadTestXPU`) needs to be called to get it member `EventHandler`, while in this period, the input class in argument cls is `CppThreadTest`, which is not defined any more. We cannot detect which instantiated class is being created in `setUpClass`, so duplicate here for enabling xpu test cases.